apps/molgenis-components/src/client/client.ts

```diff
@@ -391,31 +380,11 @@ const toFormData = (rowData: IRow) => {
 };
 
 async function convertRowToPrimaryKey(
-  myAxios: Axios,
   row: IRow,
   tableName: string,
   schemaName?: string
 ): Promise<Record<string, any>> {
-  async function getKeyValue(
-    cellValue: any,
-    column: IColumn,
-    schemaName: string
-  ) {
-    if (typeof cellValue === "string") {
-      return cellValue;
-    } else {
-      if (column.refTable) {
-        return await convertRowToPrimaryKey(
-          myAxios,
-          cellValue,
-          column.refTable,
-          schemaName
-        );
-      }
-    }
-  }
-
```

Comment:
getKeyValue is now exposed outside the convertRowToPrimaryKey contact , it that what we want ? Suggest to spit into separate PR.

Reply:
I don't know why it got moved in as an inner function in the first place. I would really prefer not to use inner functions as they can be mighty confusing (at least to me). We already had some confusion about this particular one last Friday. 
It is also only exposed within the file, since it is not exported, which is fine with me.

Reply:
imo inner function are an important/useful part of js and good to keep a small scope ( file scope tends to get to big, see code base :) ) . suggest to remove this change from this PR ( its about the external axios lib). 

About the axios lib , i do think it is a good idea/setup to inject dependancies ( in this case the lowlevel fetch client ) this allow for decoupling and reuse ( for example have the same client be used by spa ( browser runtime ) and ssr ( node runtime)

Reply:
Ok, put it back in for now :( but at the bottom of the function, so we don't get confused again why there is no return and also to improve the reading flow.